pc/storage/kdtree/kdtree.go

```diff
@@ -16,6 +16,10 @@ type KDTree struct {
 	root *node
 
 	poolNodeArray *sync.Pool
+
+	// Squared distance to cut the search branch.
+	// If this value is larger than zero, Nearest will be approximated search.
+	MinDistSq float32
```

Comment:
Do you think it is better to make it a property of the tree more than an argument of the `Nearest` method? I guess one can also set `MinDistSq` to 0 if the original behavior of `Nearest` is needed without rebuilding the tree.

Reply:
I thought it, but it makes `MinDistSq` general parameter of `storage.Search` interface.
`MinDistSq` is kd-tree (some other may have same) specific parameter, so I don't want to put it to general `storage.Search`.

Reply:
It makes sense.

Reply:
Added `KDTree.With(...KDTreeOption)` method to create shallow copy of the tree with different options.
https://github.com/seqsense/pcgol/pull/29/files#diff-406022fe6d5d2a5bc87edb91256ccbb800bd4f4eff7efcd36dd3b3a0eb3741dbR157-R161